### PR TITLE
[Cleanup] Nuke obfuscation from orbit

### PR DESCRIPTION
**Pull request by Fuzzbawls**
"It's the only way to be sure!"

This PR removes the long-unused obfuscation classes, their related files, as well as all implementations throughout our codebase.

### CODE REVIEW

File: src/masternode.cpp [hunk 1]
```diff
@@ -206,18 +210,16 @@ void CMasternode::Check(bool forceCheck)
...
-        tx.vin.push_back(vin);
-        tx.vout.push_back(vout);
-
         {
             TRY_LOCK(cs_main, lockMain);
-            if (!lockMain) return;
-
-            if (!AcceptableInputs(mempool, state, CTransaction(tx), false, NULL)) {
-                activeState = MASTERNODE_VIN_SPENT;
+            if(!lockMain) return;
```
Comment (random-zebra) [inline src/masternode.cpp:215]: nit: space `if (`

File: src/masternode.cpp [hunk 2]
```diff
@@ -206,18 +210,16 @@ void CMasternode::Check(bool forceCheck)
...
+
+            CCoins coins;
+            if(!pcoinsTip->GetCoins(vin.prevout.hash, coins) ||
```
Comment (random-zebra) [inline src/masternode.cpp:218]: nit: space `if (`

File: src/masternode.cpp [hunk 3]
```diff
@@ -624,9 +620,11 @@ bool CMasternodeBroadcast::CheckInputsAndAdd(int& nDoS)
             return false;
         }
 
-        if (!AcceptableInputs(mempool, state, CTransaction(tx), false, NULL)) {
-            //set nDos
-            state.IsInvalid(nDoS);
+        CCoins coins;
+        if(!pcoinsTip->GetCoins(vin.prevout.hash, coins) ||
```
Comment (random-zebra) [inline src/masternode.cpp:624]: nit :  space `if (`

**Comment by random-zebra** [comment]
This looks good so far.

`ThreadCheckObfuScationPool` seems to be only used to check that a masternode collateral remains unspent. How about renaming it?

Also, there are additional removals that would fit well here.

1) `CMasternodeMan::nDsqCount`: might require some trick ro remove it from the serialization. But could already be nuked from `CMasternodeMan::ToString()`

2) `CCoinControl::useObfuScation` never used.

3) `strObfuscationPoolDummyAddress` consensus param not used anymore.

4) `CWallet::GenerateObfuscationOutputs` function signature remnant.

5) RPC `getmasternodecount` mentions `obfcompat` (obfuscation compatible masternode). Can be removed.

6) Comments to `fLiteMode` still mentioning obfuscation can be updated.

**Comment by Fuzzbawls** [comment]
Updated.

the removal of `CMasternodeMan::nDsqCount` from serialization can be done in a separate PR, other comments have been addressed.

**Comment by furszy** [comment]
Needs rebase

File: src/main.cpp [hunk 4]
```diff
@@ -5825,43 +5807,7 @@ bool static ProcessMessage(CNode* pfrom, std::string strCommand, CDataStream& vR
         std::vector<unsigned char> vchSig;
         int64_t sigTime;
```
Comment (random-zebra) [inline src/main.cpp:5808]: this variable is no longer used.

Comment (Fuzzbawls) [reply src/main.cpp]: removed

**Pull request merged**